x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/security/authz_api_keys/10_index_authz.yml

```diff
@@ -0,0 +1,343 @@
+  - do:
+      catch: forbidden
+      headers:
+        Authorization: "ApiKey ${credentials}"
+      index:
+        id: "c4"
+        index: only_create_doc
+        body: >
+          {
+            "name" : "create-doc-4"
+          }
+
+  # Error: Cannot update docs in an index with only create_doc privileges
+  - do:
+      catch: forbidden
+      headers:
+        Authorization: "ApiKey ${credentials}"
+      update:
+        id: "c1"
+        index: only_create_doc
+        body: >
+          {
+            "doc": { "name" : "updated-create-doc-1" }
+          }
+
+  # Subset of the above, via bulk
+  - do:
+      headers:
+        Authorization: "ApiKey ${credentials}"
+      bulk:
+        body:
+          - '{"index": {"_index": "only_read", "_id": "r3"}}'
+          - '{"name": "read-doc-3"}'
+          - '{"create": {"_index": "only_delete"}}'
+          - '{"name": "delete-doc-3"}'
+          - '{"update": {"_index": "only_create_doc", "_id": "c2"}}'
+          - '{"doc": {"name": "updated-create-doc-2"}}'
+  - match: { errors: true }
+  - match: { items.0.index.status: 403 }
+  - match: { items.0.index.error.type: "security_exception" }
+  - match: { items.1.create.status: 403 }
+  - match: { items.1.create.error.type: "security_exception" }
+  - match: { items.2.update.status: 403 }
+  - match: { items.2.update.error.type: "security_exception" }
+
+  - do:
+      indices.refresh:
+        index: only_create_doc,only_read,only_delete
+
+  - do:
+      search:
+        index: only_create_doc,only_read,only_delete
+  - match: { hits.total.relation: "eq" }
+  - match: { hits.total.value: 3 } # Existing docs
+
+  ###
+  # Mixed success / failure
+  ###
+
+  - do:
+      headers:
+        Authorization: "ApiKey ${credentials}"
+      bulk:
+        body:
+          - '{"index": {"_index": "only_read", "_id": "r4"}}'
+          - '{"name": "read-doc-4"}'
+          - '{"index": {"_index": "only_index", "_id": "i4"}}'
+          - '{"name": "index-doc-4"}'
+          - '{"create": {"_index": "only_create_doc" }}'
+          - '{"name": "create-doc-5"}'
+  - match: { errors: true }
+  - match: { items.0.index.status: 403 }
+  - match: { items.0.index.error.type: "security_exception" }
+  - match: { items.1.index.status: 201 }
+  - match: { items.1.index._id: "i4" }
+  - match: { items.2.create.status: 201 }
+  - is_true: items.2.create._id
+
+  - do:
+      indices.refresh:
+        index: only_create_doc,only_read,only_index
+
+  - do:
+      search:
+        index: only_create_doc
+  - match: { hits.total.relation: "eq" }
+  - match: { hits.total.value: 4 } # 3 existing + 1 new
+
+  - do:
+      search:
+        index: only_read
+  - match: { hits.total.relation: "eq" }
+  - match: { hits.total.value: 0 }
+
+  - do:
+      search:
+        index: only_index
+  - match: { hits.total.relation: "eq" }
+  - match: { hits.total.value: 4 } # 3 existing + 1 new
```

Comment:
nit: should there also a test that searches the indices with the API key ?

Reply:
At some point, but this is an indexing testing not a search test.